include/cinder/GeomIo.h

```diff
@@ -559,12 +559,65 @@ class Torus : public Source {
 	int			mNumRings, mNumAxis;
 };
 
+class TorusKnot : public Source {
+public:
+	TorusKnot();
+
+	// Enables colors. Disabled by default.
+	TorusKnot&	colors( bool enable = true ) { mHasColors = enable; return *this; }
+	TorusKnot&	subdivisionsAxis( int subdiv ) { mSubdivisionsAxis = subdiv; return *this; }
+	TorusKnot&	subdivisionsHeight( int subdiv ) { mSubdivisionsHeight = subdiv; return *this; }
+	//! Specifies the \a P and \a Q parameters of the torus knot.
+	TorusKnot&	parameters( int p, int q ) { mP = p; mQ = q; return *this; }
+	//! Specifies the radius of the tube.
+	TorusKnot&	radius( float radius ) { mRadius = math<float>::max( 0, radius ); return *this; }
+	//! Allows you to scale the generated curve.
+	TorusKnot&	scale( const vec3 &scale ) { mScale = scale; return *this; }
+	//! Allows you to scale the generated curve.
+	TorusKnot&	scale( float x, float y, float z ) { mScale = vec3( x, y, z ); return *this; }
+	
+	size_t		getNumVertices() const override;
+	size_t		getNumIndices() const override;
+	Primitive	getPrimitive() const override { return Primitive::TRIANGLES; }
+	uint8_t		getAttribDims( Attrib attr ) const override;
+	AttribSet	getAvailableAttribs() const override;
+	void		loadInto( Target *target, const AttribSet &requestedAttribs ) const override;
+	TorusKnot*	clone() const override { return new TorusKnot( *this ); }
+
+protected:
+	void		calculate( std::vector<vec3> *positions, std::vector<vec3> *normals, std::vector<vec2> *texCoords, std::vector<vec3> *colors, std::vector<vec3> *tangents, std::vector<uint32_t> *indices ) const;
+
+	inline int	gcd( int a, int b ) const
```

Comment:
I agree it is a nice function to have in Cinder anyway, but since I only needed it here, and I didn't want to clutter ci::math<> (as I don't know how long it will be around), I decided to inline it in the class itself.